### PR TITLE
Add sudo troubleshooting guide

### PR DESCRIPTION
**Pull request by pbrezina**

### CODE REVIEW

File: src/troubleshooting/sudo.rst [hunk 1]
```diff
@@ -1,12 +1,332 @@
...
+  sudo`` or similar.
+- In ``/etc/sssd/sssd.conf`` check that ``sudo provider is enabled``.
+  Look at ``[domain]`` section, sudo provider is always enabled for ldap,
+  ad and ipa providers, unless this section contains
+  ``sudo_provider = none``.
+
+Obtaining logs
+**************
+
+Logs can provide many useful information when finding a solution for
```
Comment (justin-stephenson) [inline src/troubleshooting/sudo.rst:27]: Suggested rewording

```
Logs can provide useful information to find a solution when troubleshooting an issue.
```

File: src/troubleshooting/sudo.rst [hunk 2]
```diff
@@ -1,12 +1,332 @@
...
+your troubles.
+
+How do I get sudo logs?
+=======================
+
+#. Open ``/etc/sudo.conf`` and put down the following lines:
```
Comment (justin-stephenson) [inline src/troubleshooting/sudo.rst:33]: s/put down/add/

File: src/troubleshooting/sudo.rst [hunk 3]
```diff
@@ -1,12 +1,332 @@
...
+      [sssd[sudo]] [sudosrv_get_sudorules_from_cache] (0x0400): Returning $num-rules rules for [user-1@LDAP.PB]
+
+- What filter was used to fetch rules from cache?
+
+  .. code-block::
+
+      [sudosrv_get_sudorules_query_cache] (0x0200): Searching sysdb with [(&(objectClass=sudoRule)(|(sudoUser=ALL)(sudoUser=user-1)(sudoUser=#10001)(sudoUser=%group-1)(sudoUser=%user-1)(sudoUser=+*)))]
+
+- You can then use this filter to lookup in SSSD cache to see what
+  rules were returned
```
Comment (justin-stephenson) [inline src/troubleshooting/sudo.rst:81]: s/lookup in/search in the/

File: src/troubleshooting/sudo.rst [hunk 4]
```diff
@@ -1,12 +1,332 @@
...
+
+          $ dnf install ldb-tools
+          $ ldbsearch -H /var/lib/sss/db/cache_$domain.ldb -b cn=sysdb '$filter'
+
+      .. rhel-tab::
+
+          $ yum install ldb-tools
+          $ ldbsearch -H /var/lib/sss/db/cache_$domain.ldb -b cn=sysdb '$filter'
+
+- SSSD cache uses LDAP-like format equal to sudo format described in
```
Comment (justin-stephenson) [inline src/troubleshooting/sudo.rst:95]: SSSD cache uses a LDAP-like format equivalent to the sudo format described in

File: src/troubleshooting/sudo.rst [hunk 5]
```diff
@@ -1,12 +1,332 @@
...
+
+       .. code-tab::
+           :label: GSSAPI
+
+           $ ldapsearch -Y GSSAPI -H ldap://ldap.example.com -b dc=ldap,dc=example,dc=com '$filter'
+
+sudo debug logs -- sudo_debug
+=============================
+
+-  Information about user that is attempting to run sudo
```
Comment (justin-stephenson) [inline src/troubleshooting/sudo.rst:148]: s/user/the user/

File: src/troubleshooting/sudo.rst [hunk 6]
```diff
@@ -1,12 +1,332 @@
...
+
+-  SSSD is returning rules
+
+   .. code-block::
+
+       Mar 31 16:11:15 sudo[22259] Received 3 rule(s)
+
+-  ...and sudo is evaluating them by matching sudoHost, sudoUser, ...
+   attributes to current user
+-  hostname is a match
```
Comment (justin-stephenson) [inline src/troubleshooting/sudo.rst:199]: Could be single bullet here?

File: src/troubleshooting/sudo.rst [hunk 7]
```diff
@@ -1,12 +1,332 @@
...
+but ``sudo -l`` command that lists all rules still requires
+authentication. If you want ``sudo -l`` to be password-less you need to
+set ``!authenticate`` also in ``cn=defaults``.
+
+It takes too long to update rules
+=================================
+
+Look at ``man sssd-sudo`` to see how sudo rules are cached in SSSD.
+
+What is alternative to options in command definition in sudoers
```
Comment (justin-stephenson) [inline src/troubleshooting/sudo.rst:234]: s/What is// ?

**Comment by pbrezina** [comment]
Done. Please, see the changes.

File: src/troubleshooting/sudo.rst (hunk 6 above)
Comment (pbrezina) [reply src/troubleshooting/sudo.rst]: Changed it to:
```
-  ...and sudo is evaluating them by matching sudoHost, sudoUser, ...
   attributes to current user (the hostname is matched in this example)
```

**Comment by justin-stephenson** [comment]
Ack.

**Pull request merged**